core/src/main/scala/sttp/tapir/annotations/package.scala

```diff
@@ -0,0 +1,8 @@
+package sttp.tapir
+
+import sttp.tapir.generic.internal.RequestAnnotations
+
+package object annotations {
+  def deriveEndpointInput[A]: EndpointInput[A] =
```

Comment:
would it be possible to only keep this way of using the annotated model? While annotations are good for driving macros, macro-annotations (with the paradise plugin) have two significant drawbacks:

* they add members to the companion object, which is surprising and often not clear why a member that is not explicitly there in code can be referenced
* they are a dead-end, since they are not supported in Dotty/Scala3 in any form

Reply:
Sorry, need to clarify this. Do you want to remove macro-annotation `@tapirrequest` at all? Or you just want to make sure this annotation is optional and user may use only method `deriveEndpointInput`?

Reply:
No, I'd like to keep the annotation - it's needed to specify the path, for example (as we discussed below), but I suppose it would be optional?

I don't want to use macro-paradise or macro annotations (in this sense: https://docs.scala-lang.org/overviews/macros/annotations.html), as they are a construct that won't be supported in any way in Scala3. So you would have to always use `deriveEndpointInput`.

What do you think?

Reply:
Initially I wanted to specify path in argument for `deriveEndpointInput`. But let's use annotation `@endpointInput`.

I removed macro paradise plugin.